### ISSUE TITLE
Digraph combining functions modify a list digraph given as an argument

### ISSUE THREAD

**Issue by wilfwilson** [digraphs/Digraphs#467]
According to the documentation, you can either call `DigraphEdgeUnion` with a list of digraphs, or with one digraph per argument, i.e. `DigraphEdgeUnion(D1, D2, ...)` or `DigraphEdgeUnion([D1, D2, ...])`

In the second version, the list is changed in-place. This should not happen.
```
gap> D1 := CycleDigraph(3);; D2 := DigraphReverse(D1);; L := [D1, D2];
[ <immutable cycle digraph with 3 vertices>,
  <immutable digraph with 3 vertices, 3 edges> ]
gap> DigraphEdgeUnion(L) = CompleteDigraph(3);  # correct answer
<immutable digraph with 3 vertices, 6 edges>
gap> D1;  # unchanged
<immutable cycle digraph with 3 vertices>
gap> D2;  # unchanged
<immutable digraph with 3 vertices, 3 edges>
gap> L;  # wrongly changed
[ [ [ 2, 3 ], [ 3, 1 ], [ 1, 2 ] ], [ [ 3 ], [ 1 ], [ 2 ] ] ]
```
This has the consequence that if the list you give is immutable, then an error is given:
```
gap> D1 := CycleDigraph(3);; D2 := DigraphReverse(D1);; L := Immutable([D1, D2]);;
gap> IsMutable(L);
false
gap> CallFuncList(DigraphEdgeUnion, L) = CompleteDigraph(3);  # still correct in the 'multiple arguments' form
true
gap> DigraphEdgeUnion(L);  # bad
Error, List Assignment: <list> must be a mutable list in
  OutNeighboursMutableCopy( arg[1]
 ) at /Users/Wilf/gap/pkg/digraphs/gap/oper.gi:403 called from
DIGRAPHS_CombinationOperProcessArgs( arg
 ); at /Users/Wilf/gap/pkg/digraphs/gap/oper.gi:502 called from
<function "DigraphEdgeUnion">( <arguments> )
 called from read-eval loop at *stdin*:72
type 'quit;' to quit to outer loop
```

**Comment by wilfwilson** [comment]
In addition to `DigraphEdgeUnion`, also `DigraphJoin`, `DigraphCartesianProduct`, `DigraphDisjointUnion`, `DigraphCartesianProduct`, and `DigraphDirectProduct` have this problem.

This is because `DIGRAPHS_CombinationOperProcessArgs` changes its argument in place.

**Comment by wilfwilson** [comment]
Resolved in v1.4.1.
